cmd/miner/main.go

```diff
@@ -28,6 +26,8 @@ func main() {
 		return
 	}
 
+	runtime.GOMAXPROCS(cfg.MaxProcs)
```

Comment:
let's add a check to assert `MaxProcs` is less than or equal to `runtime.NumCPU()`. If it is more, default to `runtime.NumCPU()`.

Reply:
Done